src/signTx.c

```diff
@@ -349,26 +382,56 @@ static void signTx_handleInit_ui_runStep()
 
 	UI_STEP_BEGIN(ctx->ui_step, this_fn);
 
-	UI_STEP(HANDLE_INIT_STEP_DISPLAY_DETAILS) {
-		char* header =
-		        (ctx->commonTxData.txSigningMode == SIGN_TX_SIGNINGMODE_MULTISIG_TX) ?
-		        "Multisig transaction" :
-		        "New transaction";
+	UI_STEP(HANDLE_INIT_STEP_PROMPT_SIGNINGMODE) {
+		char bodyTxt[SIZEOF("Pool operator") + 1 + SIZEOF("transaction?") + 1];
+		const uint32_t signingModeStrIndex = ctx->commonTxData.txSigningMode - SIGN_TX_SIGNINGMODE_ORDINARY_TX;
+		ASSERT(signingModeStrIndex < ARRAY_LEN(uiSigningModeName));
+		snprintf(bodyTxt, SIZEOF(bodyTxt), "%s transaction?", ((const char*)PIC(uiSigningModeName[signingModeStrIndex])));
+		ui_displayPrompt(
+		        "Start new",
+		        bodyTxt,
+		        this_fn,
+		        respond_with_user_reject
+		);
+	}
+
+	UI_STEP(HANDLE_INIT_STEP_DISPLAY_NETWORK_DETAILS) {
+		const bool usualNetworkId = ctx->commonTxData.networkId == MAINNET_NETWORK_ID || ctx->commonTxData.networkId == TESTNET_NETWORK_ID;
+		const bool usualProtocolMagic = ctx->commonTxData.protocolMagic == MAINNET_PROTOCOL_MAGIC;
+
+		if (usualNetworkId && usualProtocolMagic) {
+			// no need to display the network details
+			UI_STEP_JUMP(HANDLE_INIT_STEP_SCRIPT_RUNNING_WARNING);
+		}
 
 		ui_displayNetworkParamsScreen(
-		        header,
+		        "Network details",
 		        ctx->commonTxData.networkId, ctx->commonTxData.protocolMagic,
 		        this_fn
 		);
 	}
 
-	UI_STEP(HANDLE_INIT_STEP_CONFIRM) {
-		ui_displayPrompt(
-		        "Start new",
-		        "transaction?",
-		        this_fn,
-		        respond_with_user_reject
-		);
+	UI_STEP(HANDLE_INIT_STEP_SCRIPT_RUNNING_WARNING) {
+		if (ctx->numCollaterals == 0) {
+			UI_STEP_JUMP(HANDLE_INIT_STEP_NO_COLLATERALS_WARNING);
+		}
+		ui_displayPaginatedText("WARNING:", "Plutus script will be run", this_fn);
+	}
+
+	UI_STEP(HANDLE_INIT_STEP_NO_COLLATERALS_WARNING) {
+		const bool noCollateralsIsUnusual = ctx->commonTxData.txSigningMode == SIGN_TX_SIGNINGMODE_PLUTUS_TX;
+		if (!noCollateralsIsUnusual || ctx->numCollaterals > 0) {
```

Comment:
```suggestion
		if (!collateralExpected || ctx->numCollaterals > 0) {
```
`!noCollateralsIsUnusual` sound like a triple-negation, so is incomprehensible.